src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +346,30 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            return HttpTelemetry.IsEnabled && request.RequestUri != null ?
+                WithLogging(handler, request, cancellationToken) :
+                handler.SendAsync(request, cancellationToken);
+
+            static async Task<HttpResponseMessage> WithLogging(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Debug.Assert(request.RequestUri != null);
+                HttpTelemetry.Log.RequestStart(
+                    request.RequestUri.IdnHost,
+                    request.RequestUri.Port,
+                    request.RequestUri.Scheme,
+                    request.RequestUri.AbsolutePath,
```

Comment:
If perf is important here, `PathAndQuery` could be used instead of `AbsolutePath`, as the string allocation cost for `PathAndQuery` will be incurred by the request anyway.

Reply:
Nit:  If parameter order doesn't matter for EventSource, the order of appearance in Uri could be used: `scheme, host, port, path`

Reply:
Thx. I will incorporate these suggestions.